// @tiptap/extension-typography/dist/tiptap-extension-typography.esm.js
import { Extension } from '@tiptap/core';
import { InputRule, emDash, ellipsis, openDoubleQuote, closeDoubleQuote, openSingleQuote, closeSingleQuote } from 'prosemirror-inputrules';

const leftArrow = new InputRule(/<-$/, '←');
const rightArrow = new InputRule(/->$/, '→');
const copyright = new InputRule(/\(c\)$/, '©');
const trademark = new InputRule(/\(tm\)$/, '™');
const registeredTrademark = new InputRule(/\(r\)$/, '®');
const oneHalf = new InputRule(/1\/2$/, '½');
const plusMinus = new InputRule(/\+\/-$/, '±');
const notEqual = new InputRule(/!=$/, '≠');
const laquo = new InputRule(/<<$/, '«');
const raquo = new InputRule(/>>$/, '»');
const multiplication = new InputRule(/\d+\s?([*x])\s?\d+$/, '×');
const superscriptTwo = new InputRule(/\^2$/, '²');
const superscriptThree = new InputRule(/\^3$/, '³');
const oneQuarter = new InputRule(/1\/4$/, '¼');
const threeQuarters = new InputRule(/3\/4$/, '¾');
const Typography = Extension.create({
    name: 'typography',
    addInputRules() {
        return [
            emDash,
            ellipsis,
            openDoubleQuote,
            closeDoubleQuote,
            openSingleQuote,
            closeSingleQuote,
            leftArrow,
            rightArrow,
            copyright,
            trademark,
            registeredTrademark,
            oneHalf,
            plusMinus,
            notEqual,
            laquo,
            raquo,
            multiplication,
            superscriptTwo,
            superscriptThree,
            oneQuarter,
            threeQuarters,
        ];
    },
});

export { Typography, copyright, Typography as default, laquo, leftArrow, multiplication, notEqual, oneHalf, oneQuarter, plusMinus, raquo, registeredTrademark, rightArrow, superscriptThree, superscriptTwo, threeQuarters, trademark };
